compiler/src/main/scala/org/scalajs/core/compiler/PrepJSInterop.scala

```diff
@@ -211,6 +211,20 @@ abstract class PrepJSInterop extends plugins.PluginComponent
 
         enterOwner(OwnerKind.NonEnumScalaMod) { super.transform(modDef) }
 
+      /* Identify and fix accessor that have the shape described in issue #1899.
+       *
+       * This fix is uses an over approximation of the shape of the tree due to
+       * loss of information from bug SI-9487. As such the fix introduces a
+       * difference in semantics for new js.Object {...}. Any private val,
+       * var and def will become public, but private[this] can still be used to
+       * to make them private.
+       */
+      case ddef: DefDef if isAnonymousJSObjectAccessorMethod(ddef) =>
+        val fixedDDef = treeCopy.DefDef(ddef, ddef.mods &~ Flag.PRIVATE,
+            ddef.name, ddef.tparams, ddef.vparamss, ddef.tpt, ddef.rhs)
+        fixedDDef.symbol.resetFlag(Flag.PRIVATE)
+        transform(fixedDDef)
```

Comment:
I suggest moving this case as a first step *within* the `case vddef: ValOrDefDef if enclosingOwner is OwnerKind.RawJSType =>`, to avoid the self-call to `transform`.